pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -651,3 +707,139 @@ func getMinTime(t1, t2 time.Duration) time.Duration {
 	}
 	return t2
 }
+
+func (exp *FlowExporter) l7Listener(stopCh <-chan struct{}) {
+	if err := os.Remove(exp.l7VisibilityController.L7SuricataSocketPath); err != nil && !os.IsNotExist(err) {
+		klog.ErrorS(err, "Error removing Suricata socket")
+	}
+	if err := os.MkdirAll(filepath.Dir(exp.l7VisibilityController.L7SuricataSocketPath), 0755); err != nil {
+		klog.Fatalf("Failed to create directory %s: %v", filepath.Dir(exp.l7VisibilityController.L7SuricataSocketPath), err)
+	}
+	listener, err := net.Listen("unix", exp.l7VisibilityController.L7SuricataSocketPath)
+	if err != nil {
+		klog.ErrorS(err, "Error listening on Suricata socket")
+		return
+	}
+	defer listener.Close()
+
+	klog.InfoS("L7 Listener Server started. Listening for connections...")
+	for {
+		select {
+		case <-stopCh:
+			return
+		default:
+			conn, err := listener.Accept()
+			if err != nil {
+				klog.ErrorS(err, "Error accepting Suricata connection")
+				continue
+			}
+			go exp.handleClientConnection(conn)
+		}
+	}
+}
+
+func (exp *FlowExporter) handleClientConnection(conn net.Conn) {
+	defer conn.Close()
+	reader := bufio.NewReader(conn)
+	for {
+		buffer, err := reader.ReadBytes('\n')
+		if err != nil && err != io.EOF {
+			klog.ErrorS(err, "Error reading data", "buffer", buffer)
+			return
+		}
+		err = exp.processLog(buffer)
+		if err != nil {
+			klog.ErrorS(err, "Error while processing L7 data")
+			return
+		}
+	}
+}
+
+func (exp *FlowExporter) processLog(data []byte) error {
+	exp.l7mut.Lock()
+	defer exp.l7mut.Unlock()
+	// Check if the event type is "http"
+	if eventType := extractEventType(data); eventType != "http" {
+		return nil
+	}
+	var event JsonToEvent
+	err := json.Unmarshal(data, &event)
+	if err != nil {
+		return fmt.Errorf("Error parsing JSON data %v", data)
+	}
+	if err = exp.addOrUpdateL7EventMap(event); err != nil {
+		return fmt.Errorf("Error while adding or Updating L7 event map %v", err)
+	}
+	return nil
+}
+
+func extractEventType(data []byte) string {
+	// Define a temporary struct for extracting the event type
+	type Temp struct {
+		EventType string `json:"event_type"`
+	}
+
+	// Parse the JSON string and extract the event type
+	var temp Temp
+	err := json.Unmarshal(data, &temp)
+	if err != nil && !strings.Contains(err.Error(), "unexpected end of JSON input") {
+		klog.ErrorS(err, "Error parsing JSON for event type:", "data", data)
+		return ""
+	}
+
+	return temp.EventType
+}
+
+func (exp *FlowExporter) addOrUpdateL7EventMap(event JsonToEvent) error {
```

Comment:
should the `JsonToEvent` parameter be a pointer instead (`*JsonToEvent`), given that the struct is quite large?